src/NServiceBus.Core/MessageSession.cs

```diff
@@ -55,5 +62,7 @@ public Task Unsubscribe(Type eventType, UnsubscribeOptions unsubscribeOptions)
 
         RootContext context;
         MessageOperations messageOperations;
+
+        internal const string SubscribeAllFlagKey = "NServiceBus.SubscribeAllFlag";
```

Comment:
given `SubscribeAll` there isn't really a use case where this can be invoked by the user. The only caller is AutoSubscribe atm. If it will become a public API, yes it would make sense (assuming we also keep the single Subscribe API). The only case I could see is where users have to migrate existing behaviors for the subscribe pipeline and need to handle things differently but I can't come up with a specific case yet so I think it's YAGNI for now.

Reply:
Good, thanks!